Лабораторная работа №1: Поле с точками в SVG

Kriw-Kriw opened the pull request:
ФИО: Кривов Д.А. 

Видео работы:
https://github.com/user-attachments/assets/00149dc4-ae49-4768-9e43-3f46e9b3b7b6

Задание:
Реализовать компонент, состоящий из SVG элемента, который образует сетку (поле) по заданным размерам в пикселях.
Компонент получает на вход (пропс) набор точек (массив) с параметрами (каждый элемент массива объект с полями x, y, color - координаты и hex цвет). Отобразить точки в видимой области, считать поле декартовым с начальной позицией в левом нижнем углу с координатами (0, 0). Компонент получает пропс с размерами поля. Компонент получает пропс с радиусом точки в пикселях.

Kriw-Kriw committed:
Add files via upload

reznikovkg commented on vue/src/components/gridSVG/GridSVG.vue:
```diff
@@ -0,0 +1,48 @@
+    },
+    createLines() {
+      const lines = [];
+      for (let i = 1; i < this.dataPoints.length; i++) {
```
Если что-то перебирается, то нужно использовать forEach

reznikovkg commented on vue/src/components/gridSVG/GridSVG.vue:
```diff
@@ -0,0 +1,48 @@
+    createLines() {
+      const lines = [];
+      for (let i = 1; i < this.dataPoints.length; i++) {
+        lines.push({
```
Видимо, лучше в целом заменить на логику с map

reznikovkg commented on vue/src/components/gridSVG/GridSVG.vue:
```diff
@@ -0,0 +1,48 @@
+<script>
+export default {
+  name: "GridSVG",
+  props: ["canvasWidth", "canvasHeight", "dataPoints", "radius"],
```
Пропсы описывать в полном стиле. Уже понятно, что занятия вы не посещали в принципе, пересмотреть все лекции и писать код по требованиям

reznikovkg commented on vue/src/components/gridSVG/GridSVG.vue:
```diff
@@ -0,0 +1,48 @@
+    <g v-for="(line, idx) in createLines()" :key="'line-' + idx">
+      <line v-bind="line" />
+    </g>
+    <g v-for="(circle, idx) in createCircles()" :key="'circle-' + idx">
```
Нельзя вызывать методы в шаблоне, используйте computed свойство

reznikovkg commented:
Подход в общем верный, но ошибки по коду и оформлению. В целом, считаем, что лаба сдана, а все ошибки исправить в следующей, тк задача продолжается далее.
Вторая задача, подержать возможность задачи двух и более наборов точек, также соединяющихся линиями.
Добавить поповеры при наведении на точку (в поповере показывать значение x/y)

Kriw-Kriw commented:
@reznikovkg, а между наборами точек должно быть какое-либо соединение? 
Или они должны независимо друг от друга существовать?

reznikovkg commented:
> @reznikovkg, а между наборами точек должно быть какое-либо соединение?
> Или они должны независимо друг от друга существовать?

Независимо, они могут накладывать друг на друга и прочее. Это просто отрисовка нескольких наборов на одной плоскости

Kriw-Kriw committed:
ЛР №2: Группы полей с точками в SVG

1. Были внесены правки с точки зрения оформления кода: 
- Пропсы описаны в полном стиле;
- Методы не вызываются в шаблоне.
2. CreateLines сменил логику с for на логику с map и стал просто Lines.
3. Теперь выводятся независимые друг от друга группы точек, а также поповеры при наведении на точку (с координатами).

Kriw-Kriw commented:
ФИО: Кривов Д.А.

Видео работы:
https://github.com/user-attachments/assets/310ed385-86ed-41a2-9b41-cf30facfbcae

Задание:
1. Были внесены правки с точки зрения оформления кода: 
- Пропсы описаны в полном стиле;
- Методы не вызываются в шаблоне.
2. CreateLines сменил логику с for на логику с map и стал просто Lines.
3. Теперь выводятся независимые друг от друга группы точек, а также поповеры при наведении на точку (с координатами).

reznikovkg commented:
Сдано

Kriw-Kriw committed:
ЛР №3: Игра Алхимик 

ФИО: Кривов Д.А.

Задача: 
Необходимо реализовать крафт-игру “Алхимик”. Наверняка знаете или слышали.
Есть список открытых предметов (будем выводить в верхней части экрана 80% поля сеткой). Рассчитываем под мобильный экран. В нижней части стол. С поля выбираются предметы на стол (кликом). На столе можно увеличитьва их количество (сложный вариант). Справа от стола, две кнопки, сбросить и смешать. Если формула предметов подобрана верно, то открывается новый предмент на стол, с которым можнно дальше смешивать.

Kriw-Kriw commented:
ФИО: Кривов Д.А.

Видео работы:
https://github.com/user-attachments/assets/48790aec-9b0e-4a92-826f-c39db5f79b8b

Задача: 
Необходимо реализовать крафт-игру “Алхимик”. Наверняка знаете или слышали.
Есть список открытых предметов (будем выводить в верхней части экрана 80% поля сеткой). Рассчитываем под мобильный экран. В нижней части стол. С поля выбираются предметы на стол (кликом). На столе можно увеличитьва их количество (сложный вариант). Справа от стола, две кнопки, сбросить и смешать. Если формула предметов подобрана верно, то открывается новый предмент на стол, с которым можнно дальше смешивать.
